rust/agama-lib/share/profile.schema.json

```diff
@@ -251,22 +251,156 @@
       }
     },
     "storage": {
-      "description": "Storage settings",
+      "title": "Storage settings",
       "type": "object",
+      "additionalProperties": false,
       "properties": {
-        "bootDevice": {
-          "description": "Device used for booting (e.g., '/dev/sda'). By default, all file systems are created in the boot device.",
-          "type": "string"
-        },
-        "lvm": {
-          "description": "Whether LVM is used.",
-          "type": "boolean"
-        },
-        "encryptionPassword": {
-          "description": "If set, the devices are encrypted using the given password.",
-          "type": "string"
+        "guided": {
+          "title": "Settings to execute a Guided Proposal",
+          "type": "object",
+          "additionalProperties": false,
+          "properties": {
+            "target": {
+              "title": "Target device",
+              "description": "Can be 'disk', 'newLvmVg' or an object with exactly one key-value pair",
+              "type": ["string", "object"],
+              "additionalProperties": false,
+              "properties": {
+                "disk": { "type": "string" },
+                "newLvmVG": { "type": "string" }
+              }
+            },
+            "boot": {
+              "title": "Configuration of the boot settings",
+              "type": "object",
+              "additionalProperties": false,
+              "properties": {
+                "configure": {
```

Comment:
Should configure be required? Or could we assume `true` by default?

Reply:
I though we assumed "true" by default.